File: front/src/api/testplanService.js
// api/testplanService.js
import { http } from '@/utils/request'

// 从localStorage获取当前项目ID
const getCurrentProjectId = () => {
  try {
    const storedProject = localStorage.getItem('currentProject');
    if (storedProject) {
      const projectData = JSON.parse(storedProject);
      return projectData.id || projectData.projectId || null;
    }
  } catch (error) {
    console.error('获取项目ID失败:', error);
  }
  return null;
};

export const testplanApi = {
  // 获取所有测试计划列表
  getTestplans: (params) => {
    // 确保params是对象
    params = params || {};

    // 获取当前项目ID
    const projectId = getCurrentProjectId();
    console.log('测试计划API - 当前项目ID:', projectId);

    // 构造干净的查询参数对象
    const cleanParams = {
      page: params.page || 1, // 默认值为 1
      pageSize: params.pageSize || 10, // 默认值为 10
      projectId: params.projectId || projectId // 优先使用传入的项目ID，否则使用当前项目ID
    };

    // 仅在 status 有值时添加
    if (params.status) {
      cleanParams.status = params.status;
    }

    console.log('测试计划API - 请求参数:', cleanParams);

    return http.get('/api/testplans/list', { params: cleanParams });
  },

  // 获取单个测试计划详情
  getTestplan(id) {
    const projectId = getCurrentProjectId();
    return http.get(`/api/testplans/${id}`, {
      params: { projectId }
    });
  },

  // 创建新测试计划
  createTestplan(data) {
    // 添加项目ID
    if (!data.projectId) {
    const projectId = getCurrentProjectId();
    data.projectId = projectId;
    }
    console.log('创建测试计划 - 数据:', data);

    return http.post('/api/testplans', data);
  },

  // 更新测试计划信息
  updateTestplan(id, data) {
    // 确保有项目ID
    if (!data.projectId) {
      const projectId = getCurrentProjectId();
      data.projectId = projectId;
    }

    return http.put(`/api/testplans/${id}`, data);
  },

  // 删除测试计划
  deleteTestplan(id) {
    const projectId = getCurrentProjectId();
    return http.delete(`/api/testplans/${id}`, {
      params: { projectId }
    });
  },

  // 获取测试计划关联的测试用例
  getAssociatedTestCases(planId) {
    return http.get(`/api/testplans/${planId}/cases`);
  },

  // 关联测试用例到测试计划
  associateTestCases(planId, caseIds) {
    return http.post(`/api/testplans/${planId}/cases`, { caseIds });
  },

  // 从测试计划中移除测试用例
  removeTestCase(planId, caseId) {
    return http.delete(`/api/testplans/${planId}/cases/${caseId}`);
  }
}

export default testplanApi

File: verityx/src/main/java/com/example/verityx/service/TestPlanService.java
package com.example.verityx.service;

import com.example.verityx.entity.TestCase;
import com.example.verityx.entity.TestPlan;
import com.example.verityx.entity.TestPlanCaseRel;

import java.util.List;

public interface TestPlanService {

    TestPlan createTestPlan(TestPlan testPlan);

    TestPlan getTestPlanById(Integer planId);

    List<TestPlan> getAllTestPlans();

    List<TestPlan> getTestPlansByProjectId(Integer projectId);

    List<TestPlan> getTestPlansWithPagination(int pageSize, int offset, String status, Integer projectId);

    int getTestPlanCount(String status, Integer projectId);

    boolean updateTestPlan(TestPlan testPlan);

    boolean deleteTestPlan(Integer planId);

    /**
     * 关联测试用例到测试计划
     *
     * @param planId 测试计划ID
     * @param caseIds 测试用例ID列表
     * @return 是否成功
     */
    boolean associateTestCases(Integer planId, List<Integer> caseIds);

    /**
     * 从测试计划中移除测试用例
     *
     * @param planId 测试计划ID
     * @param caseId 测试用例ID
     * @return 是否成功
     */
    boolean removeTestCase(Integer planId, Integer caseId);

    /**
     * 获取测试计划关联的测试用例列表
     *
     * @param planId 测试计划ID
     * @return 测试用例列表
     */
    List<TestCase> getAssociatedTestCases(Integer planId);

    /**
     * 获取测试计划关联的测试用例关系列表
     *
     * @param planId 测试计划ID
     * @return 关联关系列表
     */
    List<TestPlanCaseRel> getTestPlanCaseRels(Integer planId);
}
